lib/northbound.c

```diff
@@ -1528,6 +1569,9 @@ static int nb_callback_configuration(struct nb_context *context,
 	union nb_resource *resource;
 	int ret = NB_ERR;
 
+	if (CHECK_FLAG(nb_node->flags, F_NB_NODE_IGNORE_CBS))
+		return NB_OK;
```

Comment:
This check here is done prior to calling `nb_callback_{create,modify,destroy,move}`, then in each of those called functions the check is done again. That secpmd redundant check should just be an assert.

However, I don't know why this code is even being invoked in the first place for a module with no callbacks -- as mentioned in an earlier comment. This comment applies to the other top level checks (pre_validate, apply_finish etc..)

Reply:
nb_init() which is invoked by far_init calls nb_validate_callbacks() for all FRR daemons. These changes tries to avoid that for YANG modules that are marked with 'ignore_cbs'.